spacy/tokens/span_group.pyx

```diff
@@ -261,11 +263,22 @@ cdef class SpanGroup:
         """
         if doc is None:
             doc = self.doc
+        if doc is self.doc:
+            spans = list(self)
+        else:
+            spans = [doc.char_span(span.start_char, span.end_char, label=span.label_, kb_id=span.kb_id, span_id=span.id) for span in self]
+            for i, span in enumerate(spans):
+                if span is None:
+                    raise ValueError(Errors.E1052.format(i=i))
+                if span.kb_id in self.doc.vocab.strings:
+                    doc.vocab.strings.add(span.kb_id_)
+                if span.id in span.doc.vocab.strings:
+                    doc.vocab.strings.add(span.id_)
```

Comment:
Could you elaborate on why this is necessary?

Reply:
It's somewhat related to this: #12493

The problem is that we haven't enforced that `kb_id` or `span_id` are strings in the past (as with `label`), so we need to add the strings to the stringstore if they are available, but not crash if there are no strings associated with them.